packages/core/src/components/graphs/bar-stacked.ts

```diff
@@ -292,6 +294,27 @@ export class StackedBar extends Bar {
 			});
 	}
 
+	protected getBarWidth() {
+		const options = this.getOptions();
+		if (Tools.getProperty(options, "bars", "width")) {
+			return options.bars.width;
+		}
+		const mainXScale = this.services.cartesianScales.getMainXScale();
+		const chartWidth = DOMUtils.getSVGElementSize(this.parent, {
+			useAttrs: true,
+		}).width;
+	
+		const numberOfDomainValues = this.model.getStackKeys({}).length;
```

Comment:
I'm not a big fan of this...

We can change `getStackKeys()` a bit to fix this issue

![image](https://user-images.githubusercontent.com/14989804/146264080-ccbe6591-ba61-4c11-bb98-f967e7af02fa.png)


```ts
	getStackKeys(
		{ bins = null, groups = null } = { bins: null, groups: null }
	) 
```